vector/src/main/java/im/vector/app/features/home/AvatarRenderer.kt

```diff
@@ -157,25 +159,42 @@ class AvatarRenderer @Inject constructor(private val activeSessionHolder: Active
     fun shortcutDrawable(glideRequests: GlideRequests, matrixItem: MatrixItem, iconSize: Int): Bitmap {
         return glideRequests
                 .asBitmap()
-                .let {
-                    val resolvedUrl = resolvedUrl(matrixItem.avatarUrl)
-                    if (resolvedUrl != null) {
-                        it.load(resolvedUrl)
-                    } else {
-                        val avatarColor = matrixItemColorProvider.getColor(matrixItem)
-                        it.load(TextDrawable.builder()
-                                .beginConfig()
-                                .bold()
-                                .endConfig()
-                                .buildRect(matrixItem.firstLetterOfDisplayName(), avatarColor)
-                                .toBitmap(width = iconSize, height = iconSize))
-                    }
-                }
+                .avatarOrText(matrixItem, iconSize)
                 .apply(RequestOptions.centerCropTransform())
                 .submit(iconSize, iconSize)
                 .get()
     }
 
+    @AnyThread
+    @Throws
+    fun adaptiveShortcutDrawable(glideRequests: GlideRequests, matrixItem: MatrixItem, iconSize: Int, adaptiveIconSize: Int, adaptiveIconOuterSides: Float): Bitmap {
+        return glideRequests
+                .asBitmap()
+                .avatarOrText(matrixItem, iconSize)
+                .transform(CenterCrop(), AdaptiveIconTransformation(adaptiveIconSize, adaptiveIconOuterSides))
+                .signature(ObjectKey("adaptive-icon"))
```

Comment:
caches the transformed icon under with an extra unique key, to avoid overwritten other cached images on the same room url